src/lib/Scenes/Partner/Components/PartnerShows.tsx

```diff
@@ -55,62 +54,90 @@ export const PartnerShows: React.FC<{
+        return [
+          {
+            key: "empty_state",
+            content: <TabEmptyState text="There are no shows from this gallery yet" />,
+          },
+        ]
       }
-      setFetchingNextPage(true)
-      relay.loadMore(PAGE_SIZE, error => {
-        if (error) {
-          // FIXME: Handle error
-          console.error("PartnerShows.tsx", error.message)
+
+      const result: StickyTabSection[] = []
+      if (hasRecentShows) {
+        result.push({
+          key: "recent_shows",
+          content: <PartnerShowsRail partner={partner} />,
+        })
+      }
+
+      if (partner.pastShows.edges.length) {
+        result.push({
+          key: "past_shows_header",
+          content: (
+            <Flex mb={2}>
+              <Sans size="3t" weight="medium">
+                Past shows
+              </Sans>
+            </Flex>
+          ),
+        })
+
+        // chunk needs to be even to get seamless columns
+        const chunkSize = 8
+        for (let i = 0; i <= partner.pastShows.edges.length; i += chunkSize) {
+          const chunk = partner.pastShows.edges.slice(i, i + chunkSize)
+          const actualChunkSize = chunk.length
+          result.push({
+            key: `chunk ${i}:${actualChunkSize}`,
+            content: (
+              <Flex flexDirection="row" flexWrap="wrap">
+                {chunk.map(({ node }, index) => (
+                  <ShowGridItem itemIndex={index} key={node.id} show={node} />
+                ))}
+              </Flex>
+            ),
+          })
         }
-        setFetchingNextPage(false)
-      })
-    },
-  })
+      }
 
-  const pastShows = partner.pastShows && partner.pastShows.edges
-  if (!pastShows && !hasRecentShows) {
-    return <TabEmptyState text="There are no shows from this gallery yet" />
-  }
+      return result
+    },
+    [partner.pastShows.edges, hasRecentShows, pastShows]
+  )
 
   return (
-    <Box px={2} py={3}>
-      <PartnerShowsRail partner={partner} setHasRecentShows={setHasRecentShows} />
-      {!!pastShows &&
-        !!pastShows.length && (
-          <>
-            <Sans size="3t" weight="medium">
-              Past shows
-            </Sans>
-            <Spacer mb={2} />
-            <Flex flexDirection="row" flexWrap="wrap">
-              {pastShows.map((show, index) => {
-                const node = show.node
-                return <ShowGridItem itemIndex={index} key={node.id} show={node} />
-              })}
-            </Flex>
-            {fetchingNextPage && (
-              <Box p={2} style={{ height: 50 }}>
-                <Flex style={{ flex: 1 }} flexDirection="row" justifyContent="center">
-                  <Spinner />
-                </Flex>
-              </Box>
-            )}
-            <Spacer mb={3} />
-          </>
-        )}
-    </Box>
+    <StickyTabPageFlatList
+      data={sections}
+      onEndReachedThreshold={0.5}
+      onEndReached={() => {
+        if (isLoadingMore || !relay.hasMore()) {
```

Comment:
Like i said in the PR description, we need to look into a different way to debounce these because FlatList calls these callbacks multiple times before the react render cycle has caused `isLoadingMore` to be `true`